### PR TITLE
User inDiscord Feature - UI

### PR DESCRIPTION
**Pull request by bhtibrewal**
to view the UI navigate to `/user-management`.
###  TODO:
This data will be removed after the APIs are merged on the backend in a follow-up PR 
 
image

### CODE REVIEW

File: __tests__/users/App.test.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+          status: 200,
+          contentType: 'application/json',
+          headers: {
+            'Access-Control-Allow-Origin': '*',
+            'Access-Control-Allow-Methods': 'GET, POST, PUT, DELETE, OPTIONS',
+            'Access-Control-Allow-Headers': 'Content-Type, Authorization',
+          },
+          body: JSON.stringify(filteredUsersData),
+        });
+      } else if (url === `${API_BASE_URL}/users/verified`) {
```
Comment (bhtibrewal) [inline __tests__/users/App.test.js:41]: - these urls will be fixed in the #357

File: __tests__/users/App.test.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+    await page.goto(`${BASE_URL}/user-management/`); // Replace with your app's URL
+    await page.waitForNetworkIdle();
+  });
+
+  afterAll(async () => {
+    await browser.close();
+  });
+
+  it('should render all sections', async () => {
```
Comment (bhtibrewal) [inline __tests__/users/App.test.js:55]: - will add more tests in  #357

File: __tests__/users/user-management-home-screen.test.js [hunk 3]
```diff
@@ -16,7 +16,7 @@ describe('Tests the User Management User Listing Screen', () => {
 
   beforeAll(async () => {
     browser = await puppeteer.launch({
-      headless: true,
+      headless: 'new',
```
Comment (bhtibrewal) [inline __tests__/users/user-management-home-screen.test.js:19]: puppeteer is giving warnings to use their new implementation

File: user-management/App.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  render(element, container);
+};
+
+const tabs = [
+  { display_name: 'In Discord', id: 'in_discord' },
+  { display_name: 'Linked Accounts', id: 'verified' },
+  { display_name: 'Unlinked Accounts', id: 'not_verified' },
+];
+
+// const users = await getUsers();
```
Comment (bhtibrewal) [inline user-management/App.js:20]: - TODO: will remove the dummy data and use the APIs in this follow up PR   #357

File: user-management/data.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
+// TODO: This dummy data will be removed after the APIs are merged on the backend
+export const usersData = {
```
Comment (bhtibrewal) [inline user-management/data.js:2]: - TODO: will remove the dummy data and use the APIs in this follow up PR   #357

**Review by vvaibhavdesai — DISMISSED** [review]

**Comment by vvaibhavdesai** [comment]
Very nicely done 🎸

**Review by yesyash — APPROVED** [review]

**Comment by bhtibrewal** [comment]
> can you please look into this image
> 
> I am getting this on my machine. Dimensions : 1536 x 746

yes thanks for pointing it out, it not coming on my device, will look into it

**Comment by bhtibrewal** [comment]
> image
> 
> I get these empty screens when pressing the `linked accounts` & `unlinked accounts`. Are they a work in progress? Can we comment this part out so that users cannot interact with them for now?

yes, the work is in progress, this route is not accessible to users I have not given users a button to come to this route.

**Comment by yesyash** [comment]
> > image
> > I get these empty screens when pressing the `linked accounts` & `unlinked accounts`. Are they a work in progress? Can we comment this part out so that users cannot interact with them for now?
> 
> yes, the work is in progress, this route is not accessible to users I have not given users a button to come to this route.

can you please point to the code for this ?

**Review by vvaibhavdesai — APPROVED** [review]

**Comment by bhtibrewal** [comment]
> > > image
> > > I get these empty screens when pressing the `linked accounts` & `unlinked accounts`. Are they a work in progress? Can we comment this part out so that users cannot interact with them for now?
> > 
> > 
> > yes, the work is in progress, this route is not accessible to users I have not given users a button to come to this route.
> 
> can you please point to the code for this or a ticket that handles this?
Please have a look, fixed it
image

**Review by yesyash — APPROVED** [review]

**Review by yesyash — COMMENTED** [review]
Not sure if this is just an issue with my machine, can you please try it in responsive mode by setting the dimensions to 1536 x 768

image
image

**Comment by bhtibrewal** [comment]
> y setting the dimensions to 1536

I tried but I don't see the same behaviour
image

**Comment by yesyash** [comment]
> > y setting the dimensions to 1536
> 
> I tried but I don't see the same behaviour image

The dimensions in the screenshot don't look right. Try with 1536 x 768. 
It is not a blocker and can be looked into in another ticket.

File: user-management/index.html [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    <meta charset="UTF-8" />
+    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <link rel="stylesheet" href="/style.css" />
+    <link rel="stylesheet" href="/user-management/style.css" />
+    <title>Document</title>
+  </head>
+  <body>
+    <header class="header">
+      <h1>User Management</h1>
```
Comment (bhtibrewal) [inline user-management/index.html:13]: - Route will be changed to Discord in #357

**Comment by bhavukkalra** [comment]
Just landed here, After seeing a tweet about this by the Real Dev Squad's Official Twitter handle.
Quiet amazing work here @bhtibrewal.

The one thing that caught my eye were the tests written and the TODO's on where the code can be improved even further!
Cool stuff!

**Pull request merged**